src/ipc/TypedMsgHdr.h

```diff
@@ -43,7 +44,10 @@ class TypedMsgHdr: public msghdr
     /* message type manipulation; these must be called before put/get*() */
     void setType(int aType); ///< sets message type; use MessageType enum
     void checkType(int aType) const; ///< throws if stored type is not aType
-    int type() const; ///< returns stored type or zero if none
+    MessageType type() const; ///< converts rawType() to MessageType
```

Comment:
Just a question: Other than the `messageType(hdrMsg.type())` initialization/creation use case shown in this PR diff, what are other current type() users? I wonder whenter we can completely eliminate this method...

Reply:
There are no other users. We could eliminate this method (moving its checks into StrandMessage::StrandMessage(), but it would look a little awkward IMO.

Reply:
Perhapse this should be implemented as a TypedMsgHdr conversion operator:
```
operator MessageType() { ... }
```
then the caller does this more elegant code instead:
```
 MessageType(hdrMsg)
```

I also see there are 9 calls to rawType(). It seems a bit odd to be having a bounds checked getter, but mostly ignoring it in favour of a raw getter.

Reply:
> Perhapse this should be implemented as a TypedMsgHdr conversion operator

We could do this but there is only one type() user - the benefit would be small over the fact that the implicit conversion is more difficult to control (i.e., avoid an unintended conversion).

> I also see there are 9 calls to rawType().

Right, because TypedMsgHdr is a low-level structure dealing with wire data. We should not throw upon receiving some garbage but simply discard it. There are several rawType() calls (as you noted), but they are confined within that low-level data analyzing code.

Reply:
On one hand this data is supposed to be coming from another kid of this Squid process, so any "garbage" is a sign of malfunction or bugs. Silently ignoring that is a bad idea.

On the other hand, this is I/O going via mechanisms outside of Squid. There is a possibility of third-party applications joining in on the communications. Which means trusting it unconditionally is a bad idea.

The practical middle ground is to do the bounds checking and debugs() log any unexpected incidents before otherwise ignoring them.

Reply:
>  Silently ignoring that is a bad idea.

I did not mean that we silently ignore it - in this case, a level-1 message is reported. See, e.g., Ipc::Coordinator::receive().

> The practical middle ground is to do the bounds checking and debugs()  

I don't think we need to do extra bounds checking in, e.g., Ipc::Coordinator::receive() because the values are already checked by the switch operator itself.

Reply:
I am resolving this change request because the question I asked has been answered. Thank you.

My underlying concern remains, and, while thinking about it, I will account for the ideas in the followup discussion in this change request. If I find a good way to address that concern, I will either commit the changes or post a new change request.